usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,314 @@
+                    # we update all values
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_fpds")
+                    transaction_fpds_sql = (
+                        "UPDATE transaction_fpds SET {} "
+                        "where detached_award_procurement_id = {}".format(
+                            pairs, load_object["transaction_fpds"]["detached_award_procurement_id"]
+                        )
+                    )
+                    cursor.execute(transaction_fpds_sql)
+
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_normalized")
+                    load_object["transaction_fpds"]["transaction_id"] = results[0][0]
+                    load_object["award"]["latest_tranaction_id"] = results[0][0]
+                    transaction_normalized_sql = "UPDATE transaction_normalized SET {} " "where id  = '{}'".format(
+                        pairs, load_object["transaction_fpds"]["transaction_id"]
+                    )
+                    cursor.execute(transaction_normalized_sql)
+
+                    logger.debug("updated fpds transaction {}".format(results[0][0]))
+                else:
+                    # If there is no transaction we create a new one.
+                    # transaction_normalized and transaction_fpds should be one-to-one
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_normalized")
+                    transaction_normalized_sql = (
+                        "INSERT INTO transaction_normalized {} VALUES {} " "RETURNING id".format(columns, values)
+                    )
+                    cursor.execute(transaction_normalized_sql)
+                    results = cursor.fetchall()
+                    load_object["transaction_fpds"]["transaction_id"] = results[0][0]
+                    load_object["award"]["latest_tranaction_id"] = results[0][0]
+
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_fpds")
+                    transaction_fpds_sql = (
+                        "INSERT INTO transaction_fpds {} VALUES {} " "RETURNING transaction_id".format(columns, values)
+                    )
+                    cursor.execute(transaction_fpds_sql)
+                    results = cursor.fetchall()
+
+                    logger.debug("created fpds transaction {}".format(results[0][0]))
+
+                # No matter what, we need to go back and update the award's latest transaction to the award we just made
+                update_award_lastest_transaction_sql = "UPDATE awards SET latest_transaction_id = {} where id = {}".format(
+                    results[0][0], load_object["award"]["latest_tranaction_id"]
+                )
+                cursor.execute(update_award_lastest_transaction_sql)
+
+
+def setup_load_lists(load_object, table):
+    columns = []
+    values = []
+    update_pairs = []
+    for key in OrderedDict(load_object[table]).keys():
```

Comment:
Will this give you the same order every time? The load object should already be ordered if order matters. Wrapping in an OrderedDict might be too late and not deterministic

Reply:
I'm not sure. This merits further testing and might actually trim some time off runs if we can just work with the order of the load object